include/icp/impl/vanilla_3d.h

```diff
@@ -0,0 +1,24 @@
+#include "icp/icp.h"
+
+typedef struct {
+    std::vector<float> distances;
+    std::vector<int> indices;
+} NEIGHBOR;
```

Comment:
Could you rename this to be CamelCase? Also, since it's only used for Vanilla3D, it might be a good idea to hide it under the private section.